adapters/yieldmo/yieldmo.go

```diff
@@ -35,14 +35,12 @@ type ExtBid struct {
 }
 
 func (a *YieldmoAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	var errs []error
 	var adapterRequests []*adapters.RequestData
 
 	adapterReq, errors := a.makeRequest(request)
 	if adapterReq != nil {
 		adapterRequests = append(adapterRequests, adapterReq)
 	}
-	errs = append(errs, errors...)
```

Comment:
If we remove `var errs []error`, it looks odd to not remove `var adapterRequests []*adapters.RequestData` for the same reason. Do you agree?

Maybe we can simplify to:
```
37   func (a *YieldmoAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
38 -     var adapterRequests []*adapters.RequestData
39 -
40 -     adapterReq, errors := a.makeRequest(request)
41 -     if adapterReq != nil {
42 -         adapterRequests = append(adapterRequests, adapterReq)
43 -     }
44 -
45 -     return adapterRequests, errors
   +     return a.makeRequest(request)
46   }
adapters/yieldmo/yieldmo.go
```
Whatsmore, we could even rename `makeRequest` to `MakeRequests` and remove an extra function entirely. Do you think this would be outside the scope of this PR?
```
37 - func (a *YieldmoAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
38 -     var adapterRequests []*adapters.RequestData
39 -
40 -     adapterReq, errors := a.makeRequest(request)
41 -     if adapterReq != nil {
42 -         adapterRequests = append(adapterRequests, adapterReq)
43 -     }
44 -
45 -     return adapterRequests, errors
46 - }
47  
48 - func (a *YieldmoAdapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, []error) {
   + func (a *YieldmoAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
49       var errs []error
50  
51       if err := preprocess(request); err != nil {
52           errs = append(errs, err)
53       }
54  
55       // Last Step
56       reqJSON, err := json.Marshal(request)
57       if err != nil {
58           errs = append(errs, err)
adapters/yieldmo/yieldmo.go
```

Reply:
I don't see a way to simplify `YieldmoAdapter.MakeRequests()` to a one-liner since  `a.makeRequest(request)` returns a single `*adapters.RequestData` pointer, and the `MakeRequests` method needs to return a slice of such pointers, `[]*adapters.RequestData`.   But we can have a simplified two-line version:
```go
func (a *YieldmoAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
	reqData, errors := a.makeRequest(request)
	return []*adapters.RequestData{reqData}, errors
}
```

Regarding other refactoring, I would prefer to do it in a separate / follow-up PR

Reply:
changed in 6d4f03aa

Reply:
> I don't see a way to simplify `YieldmoAdapter.MakeRequests()` to a one-liner since `a.makeRequest(request)` returns a single `*adapters.RequestData` pointer

You're right @dmitris. I didn't notice that. Thanks for the change.